artifactory/utils/docker/buildinfo.go

```diff
@@ -412,6 +418,19 @@ func (configLayer *configLayer) getNumberOfDependentLayers() int {
 	return layersNum
 }
 
+func FilterDuplicateDockerLayer(imageMLayers []layer) []layer {
```

Comment:
Looks like this function can be made private.
Also, I suggest renaming it to *removeDuplicateDockerLayers*.